src/sys/memfd.rs

```diff
@@ -40,7 +42,15 @@ libc_bitflags!(
 /// [`memfd_create(2)`]: https://man7.org/linux/man-pages/man2/memfd_create.2.html
 pub fn memfd_create(name: &CStr, flags: MemFdCreateFlag) -> Result<RawFd> {
     let res = unsafe {
-        libc::syscall(libc::SYS_memfd_create, name.as_ptr(), flags.bits())
```

Comment:
I would rather not bend over backwards to cater to the obscure arches.  If they don't define the libc symbol, then I think we should just not bind it.  We can do it like this:
`#[cfg(any(target_os = "freebsd", target_env = "gnu", target_env = "musl"))]`

Reply:
Seem that it's just armv7 uclibceabihf that's failing

Reply:
I have some solution but it's a more ugly cfg block.

Essentially use the libc symbol if it's freebsd or gnu/musl linux and otherwise use the syscall function.

Or was the intention to gate memfd_create from all linuxes that aren't using a gnu or musl libc?